CMakeLists.txt

```diff
@@ -83,6 +83,15 @@ execute_process(COMMAND mkdir -p ${CMAKE_LIBRARY_OUTPUT_DIRECTORY})
 execute_process(COMMAND mkdir -p ${CMAKE_RUNTIME_OUTPUT_DIRECTORY})
 execute_process(COMMAND ln -fsn ${CMAKE_SOURCE_DIR}/cmake/init.py ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/__init__.py)
 
+# build options
+set(ENABLE_JSON_OUTPUT "FALSE" CACHE BOOL "Enable JSON output of frames (adds to compile time and binary size)")
```

Comment:
This build option and compiler flag should just be in core/CMakeLists.txt, otherwise this will trigger rebuilding all of the other packages from scratch.

Reply:
Hmm, but there are serializable things outside of there too aren't there? (e.g. G3SkyMap looks like it's serializable).

Reply:
Yeah, but the other libraries are built against core, so wouldn’t they just inherit the right properties when linked?  I could be wrong here.

Reply:
I think the problem is that the packages would then see different versions of serialization.h and the non-core packages wouldn't generate the json serialization code at all (which is done at compile-time). 

Indeed, if I try to do this, everything builds and the tests all pass, but no evidence of json serialization:
```
$ objdump -D libspt3g-core.so | grep rapidjson | wc -l
5168
$ objdump -D libspt3g-maps.so | grep rapidjson | wc -l
0

```
and serializing a frame containing one of those to JSON fails (though less miserably than I expected... I expected to see a symbol lookup error!):

```
In [1]: from spt3g.core import G3Frame      

In [2]: frame = G3Frame()                   

In [3]: from spt3g.maps import FlatSkyMap   

In [4]: frame["map"] = FlatSkyMap()               
...
In [8]: frame.as_json()                     
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-8-ad9ef20ce3cd> in <module>
----> 1 frame.as_json()

RuntimeError: Trying to save an unregistered polymorphic type (FlatSkyMap).
Make sure your type is registered with CEREAL_REGISTER_TYPE and that the archive you are using was included (and registered with CEREAL_REGISTER_ARCHIVE) prior to calling CEREAL_REGISTER_TYPE.
If your type is already registered and you still see this error, you may need to use CEREAL_REGISTER_DYNAMIC_INIT.
```

Reply:
I see, you’re right.  I still think the definition should be in core/CMakeLists.h since all the code changes are in the core library...  I think there’s a target interface option you can use instead of a global one?  That should properly propagate the definitions to libraries linked against core.

Reply:
Ah, yes, that does look like it would work! Thanks, I didn't know about that cmake feature.